core/src/accounts_hash_verifier.rs

```diff
@@ -106,55 +106,63 @@ impl AccountsHashVerifier {
             exit,
             fault_injection_rate_slots,
+            accounts_hash,
         );
 
-        Self::submit_for_packaging(accounts_package, pending_snapshot_package, snapshot_config);
+        Self::submit_for_packaging(
+            accounts_package,
+            pending_snapshot_package,
+            snapshot_config,
+            accounts_hash,
+        );
     }
 
-    fn verify_accounts_package_hash(accounts_package: &AccountsPackage) {
+    /// returns accounts hash
+    fn verify_accounts_package_hash(accounts_package: &AccountsPackage) -> Hash {
         let mut measure_hash = Measure::start("hash");
-        if let Some(expected_hash) = accounts_package.accounts_hash_for_testing {
-            let mut sort_time = Measure::start("sort_storages");
-            let sorted_storages = SortedStorages::new(&accounts_package.snapshot_storages);
-            sort_time.stop();
+        let mut sort_time = Measure::start("sort_storages");
+        let sorted_storages = SortedStorages::new(&accounts_package.snapshot_storages);
+        sort_time.stop();
 
-            let mut timings = HashStats {
-                storage_sort_us: sort_time.as_us(),
-                ..HashStats::default()
-            };
-            timings.calc_storage_size_quartiles(&accounts_package.snapshot_storages);
-
-            let (hash, lamports) = accounts_package
-                .accounts
-                .accounts_db
-                .calculate_accounts_hash_without_index(
-                    &CalcAccountsHashConfig {
-                        use_bg_thread_pool: true,
-                        check_hash: false,
-                        ancestors: None,
-                        use_write_cache: false,
-                        epoch_schedule: &accounts_package.epoch_schedule,
-                        rent_collector: &accounts_package.rent_collector,
-                    },
-                    &sorted_storages,
-                    timings,
-                )
-                .unwrap();
-
-            assert_eq!(accounts_package.expected_capitalization, lamports);
-            assert_eq!(expected_hash, hash);
+        let mut timings = HashStats {
+            storage_sort_us: sort_time.as_us(),
+            ..HashStats::default()
+        };
+        timings.calc_storage_size_quartiles(&accounts_package.snapshot_storages);
+
+        let (accounts_hash, lamports) = accounts_package
+            .accounts
+            .accounts_db
+            .calculate_accounts_hash_without_index(
+                &CalcAccountsHashConfig {
+                    use_bg_thread_pool: true,
+                    check_hash: false,
+                    ancestors: None,
+                    use_write_cache: false,
+                    epoch_schedule: &accounts_package.epoch_schedule,
+                    rent_collector: &accounts_package.rent_collector,
+                },
+                &sorted_storages,
+                timings,
+            )
+            .unwrap();
```

Comment:
Just to make sure if I understand this PR.

So previously, we conditionally compute the accounts hash here just for verification purposes.
But now we always compute accounts hash here as AccountsPackage no longer stores accounts hash.

Reply:
yes. we used to compute it earlier and store the computed hash in AccountsPackage. Now, we wait to compute it here.